resources/analysisTools/tools/bamReadPairExtractor.pl

```diff
@@ -123,361 +124,387 @@ BEGIN
+        $selected = check_selection($fields, $flags, $noiseHash);
+        if (!$selected) {
+            $selected = check_selection($fields2, $flags2, $noiseHash);
         }
-      }
-
-
-      if (! $flags2->{UNMAPPED} && $flags2->{REVERSED}) {
-        ($fields2->[9] = reverse $fields2->[9]) =~ tr/ACGTacgt/TGCAtgca/;
-        $fields2->[10] = reverse $fields2->[10];
-        if ($flags2->{M_UNMAPPED}) {
-          ($fields->[9] = reverse $fields->[9]) =~ tr/ACGTacgt/TGCAtgca/;
-          $fields->[10] = reverse $fields->[10];
+        if (!$selected) {
+            delete($readHash->{$fields->[0]});
+            $hs--;
+            next;
+        }
+        else {
+            $matenr = (!$flags->{PAIRED}) ? 1 : ($flags->{FIRST_MATE}) ? 1 : 2;
+            $matenr2 = (!$flags2->{PAIRED}) ? 1 : ($flags2->{FIRST_MATE}) ? 1 : 2;
+            ($matenr != $matenr2) || die "Mate number clash in input line $.";
+
+            if ((!$flags->{UNMAPPED}) && $flags->{REVERSED}) {
+                ($fields->[9] = reverse $fields->[9]) =~ tr/ACGTacgt/TGCAtgca/;
+                $fields->[10] = reverse $fields->[10];
+                if ($flags->{M_UNMAPPED}) {
+                    ($fields2->[9] = reverse $fields2->[9]) =~ tr/ACGTacgt/TGCAtgca/;
+                    $fields2->[10] = reverse $fields2->[10];
+                }
+            }
+
+            if (!$flags2->{UNMAPPED} && $flags2->{REVERSED}) {
+                ($fields2->[9] = reverse $fields2->[9]) =~ tr/ACGTacgt/TGCAtgca/;
+                $fields2->[10] = reverse $fields2->[10];
+                if ($flags2->{M_UNMAPPED}) {
+                    ($fields->[9] = reverse $fields->[9]) =~ tr/ACGTacgt/TGCAtgca/;
+                    $fields->[10] = reverse $fields->[10];
+                }
+            }
+
+            #trim sequences that contain bases with quality < qthreshold
+            if (TRIMQUALITYTHRESHOLD() > 0) {
+                trim_bwa($fields, TRIMQUALITYTHRESHOLD());
+                trim_bwa($fields2, TRIMQUALITYTHRESHOLD());
+            }
+
+
+            #check if length > lthreshold
+            my $seqlength = length($fields->[9]);
+            my $seqlength2 = length($fields2->[9]);
+            if (MINREADLENGTH() > 0 && $seqlength < MINREADLENGTH() && $seqlength2 >= MINREADLENGTH()) {
+                if (FASTQ) {
+                    printf {$fq{u}} "\@%s/%s\n%s\n+\n%s\n", $fields2->[0], $matenr2, $fields2->[9], $fields2->[10];
+                }
+                if (SCARF) {
+                    printf {$sc{u}} "%s/%s\:%s\:%s\n", $fields2->[0] =~ s/[^\d:]//rg, $matenr2, $fields2->[9], $fields2->[10] =~ tr/[\x{21}-\x{5F}]/[\x{40}-\x{7E}]/r;
+                }
+                COUNT && $count{single_read_deleted}++;
+                delete($readHash->{$fields->[0]});
+                $hs--;
+                next;
+            }
+            elsif (MINREADLENGTH() > 0 && $seqlength2 < MINREADLENGTH() && $seqlength >= MINREADLENGTH()) {
```

Comment:
Why do you place elsif on a new line?

Reply:
I overlooked that this was caused by the auto-code-cleanup function in Idea.